python/pyiceberg/manifest.py

```diff
@@ -0,0 +1,155 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from enum import Enum
+from functools import singledispatch
+from typing import (
+    Any,
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Union,
+)
+
+from pyiceberg.avro.file import AvroFile
+from pyiceberg.avro.reader import AvroStruct
+from pyiceberg.io.base import InputFile
+from pyiceberg.schema import Schema
+from pyiceberg.types import (
+    IcebergType,
+    ListType,
+    MapType,
+    PrimitiveType,
+    StructType,
+)
+from pyiceberg.utils.iceberg_base_model import IcebergBaseModel
+
+
+class FileFormat(str, Enum):
+    AVRO = "AVRO"
+    PARQUET = "PARQUET"
+    ORC = "ORC"
```

Comment:
What about making these lower case? I think that's a more natural way to use them. For example, if we have a way to add an extension to a file name, these names would probably be the extension.

Reply:
Do you mean the key or the string repr? For the keys, we decided to upper-case them because that's also the convention in the standard lib. The value I just took from an example Manifest: https://github.com/apache/iceberg/blob/d1d70875714ada876cc651be6d1b3a1d7483a0da/python/tests/conftest.py#L233

Reply:
The string, but if Java is producing PARQUET, then we should probably do the same case.

Reply:
This is how it is in the Avro file:
```
➜  warehouse git:(main) avro-tools tojson nyc/taxis_sample/metadata/2ac3263d-9174-418d-b4cc-0542d4c2ea0b-m0.avro | jq | grep -i file_format
22/07/22 13:55:37 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
    "file_format": "PARQUET",
    "file_format": "PARQUET",
    "file_format": "PARQUET",
    "file_format": "PARQUET",
```